sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -0,0 +1,202 @@
+package io.sentry.android.core;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import io.sentry.DateUtils;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.ILogger;
+import io.sentry.ISerializer;
+import io.sentry.ObjectWriter;
+import io.sentry.Scope;
+import io.sentry.SentryEnvelope;
+import io.sentry.SentryEnvelopeItem;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.Session;
+import io.sentry.protocol.App;
+import io.sentry.protocol.Device;
+import io.sentry.protocol.SentryId;
+import io.sentry.protocol.User;
+import io.sentry.util.MapObjectWriter;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Sentry SDK internal API methods meant for being used by the Sentry Hybrid SDKs. */
+@ApiStatus.Internal
+public final class InternalSentrySdk {
+
+  /**
+   * @return a copy of the current hub's topmost scope, or null in case the hub is disabled
+   */
+  @Nullable
+  public static Scope getCurrentScope() {
+    final @NotNull AtomicReference<Scope> scopeRef = new AtomicReference<>();
+    HubAdapter.getInstance().withScope(scopeRef::set);
```

Comment:
Also, `withScope` has side effects and isn't intended to be used on Android, since it's a single hub/scope.
It's being even removed from a few SDKs in favor of the scope lambda -> `captureException(scope {})`
This might have side effects, I know @romtsn dislike this API and already had issues with it as well.

Reply:
This is intentional, as it returns a safe copy of the current scope. I already talked to Roman about this, this is probably the only good use-case for using `withScope` 😅

Reply:
H: `withScope` isn't thread-safe because it's meant to run in a single thread.
Android is single hub/scope.
If another thread set something on the scope while the `push` has been executed but not the `pop` yet, the data is set in the wrong scope, IMO this is still a bug, and `withScope` should never be used in Android.
This API was a mistake and is already deprecated in a few SDKs as mentioned, in favor of the `scope` lambda.

Reply:
@markushi the usage of `withScope` is still there, mind sharing a commit link or PR?

Reply:
I thought[ this PR](https://github.com/getsentry/sentry-java/pull/2846/files#diff-8a43e80beabd704bc935bf3f79710e3934bbe99dffca6b8f4a3e3a008c4f4cc5R45) solved that, but looking closer at the code the `withScope` is used.
- https://github.com/getsentry/sentry-java/pull/2846